architecture/additional_concepts/storage.adoc

```diff
@@ -492,6 +492,183 @@ spec:
+----
+
+Then, look through the file for any PersistentVolumeClaim definitions and edit
+them to include an option to specify the storage class. The MongoDB template has
+one PVC defined. The line highlighted below shows what needs to be added to allow
+specification of a non-default storage class.
+
+.PVC Non-default StorageClass example
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: ${DATABASE_SERVICE_NAME}
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: ${VOLUME_CAPACITY}
+  storageClassName: ${STORAGE_CLASS_NAME}
+----
+
+Then, at the bottom of the template file, add the new parameter
+`STORAGE_CLASS_NAME` to the parameter list.
+
+.PVC Non-default StorageClass example contd
+[source,yaml]
+----
+description: Storage class to use for persistent storage
+displayName: Storage class name
+name: STORAGE_CLASS_NAME
+required: true
+----
+
+Now that you have an updated template, you can use it from the CLI or web console.
+From the CLI, you can pass in the required parameters to `oc process` and
+create them.
+
+----
+$ oc process -f mongodb-export.yml \
+STORAGE_CLASS_NAME=gp2-encrypted-us-east-2b | oc create -f
+----
+
+Or you can upload your template via the web console and fill out the parameters
+of the template interactively. To do that, click the "Add to Project" dropdown
+in the application console and select the option "Import YAML / JSON".
+
+image::add-to-project.png[Import yaml/json to project,title="Import yaml/json to project"]
+
+From here, you can either upload your template file or copy/paste it into the
+provided text field and then click create. From here you can opt to process
+and/or save your template file to the selected namespace for future use.
+If you process the template, you'll be taken to a page to fill out the available
+template parameters, including the desired storage class. After populating the parameters,
+you can create the MongoDB service and confirm that your PVC has been created in
+the correct AZ.
+
+[[osd-multi-az-pvc-best-practices]]
+=== Best practices
+
+It is very important to deliberately select a storage class when provisioning
+PVCs in a multi-AZ OSD cluster. If every PVC is always provisioned by the
+default storage class, this can lead to problems in the future.
+
+One of these problems is an uneven distribution of workload. If you are
+primarily running workloads that make use of persistent storage, and you always
+provision that storage using the default storage class, all of your pods will be
+scheduled to nodes in that AZ. This can lead to a situation where the nodes in one
+AZ are running out of capacity while the rest of the cluster is underutilized.
+Once you are in this state, there isn't an easy method to redistribute the workloads.
+Any redistribution will require backing up and restoring the contents of PVs manually.
+The other problem with winding up in this state is that it effectively undermines
```

Comment:
s/with winding up in this state/with this state
s/goes down/fails
s/go down/be lost